File: Sprint8/I/C++/main.cpp
// https://contest.yandex.ru/contest/26131/problems/I/
#include <iostream>
#include <string>
#include <vector>

std::vector<int> prefix_function(std::string& s) {
    std::vector<int> prefix(s.length(), 0);
    for (int i = 1; i < s.length(); i++) {
        int k = prefix[i - 1];
        while (k > 0 && s[k] != s[i]) {
            k = prefix[k - 1];
        }
        if (s[k] == s[i]) {
            k += 1;
        }
        prefix[i] = k;
    }
    return prefix;
}

int main() {
    std::string s;
    std::cin >> s;

    std::vector<int> prefix = prefix_function(s);

    if (prefix[s.size() - 1] >= s.size() / 2) {
        std::cout << s.size() / (s.size() - prefix[s.size() - 1]) << std::endl;
    } else {
        std::cout << 1 << std::endl;
    }

    return 0;
}

File: Sprint8/I/Rust/src/main.rs
// https://contest.yandex.ru/contest/26131/problems/I/

fn prefix_function(s: &str) -> Vec<usize> {
    let bytes = s.as_bytes();
    let mut prefix = vec![0; s.len()];

    for i in 1..s.len() {
        let mut k = prefix[i - 1];
        while k > 0 && bytes[k] != bytes[i] {
            k = prefix[k - 1];
        }
        if bytes[k] == bytes[i] {
            k += 1;
        }

        prefix[i] = k;
    }

    prefix
}

fn main() {
    let mut s = String::new();
    std::io::stdin().read_line(&mut s).unwrap();
    let s = s.trim();

    let prefix = prefix_function(&s);

    let last_prefix_value = prefix[s.len() - 1];
    if last_prefix_value >= s.len() / 2 {
        println!("{}", s.len() / (s.len() - last_prefix_value));
    } else {
        println!("1");
    }
}
